hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -208,6 +240,8 @@ public AbfsClient(final URL baseUrl, final SharedKeyCredentials sharedKeyCredent
 
   @Override
   public void close() throws IOException {
+    runningTimerTask.cancel();
```

Comment:
Lets add a null check in case runningTimerTask is not spawned before client close.

Reply:
taken.